integration_tests/test_suites/daemon-test-suite/auto_run_reexecution_tests/test_auto_run_reexecution.py

```diff
@@ -48,6 +48,21 @@ def create_run(instance, **kwargs):
         )
 
 
+def process_runs_in_queue(instance, queue: Sequence[DagsterRun]):
+    """The MockedRunCoordinator doesn't update the status of a run when submit_run is called. All other
+    RunCoordinators move the run from NOT_STARTED to another status, so post-process the queue to update the
+    status of each run to STARTED, since the auto-reexecution logic checks the status of the runs.
+    """
+    for run in queue:
+        updated_run = instance.get_run_by_id(run.run_id)
+        if updated_run.status == DagsterRunStatus.NOT_STARTED:
+            launch_started_event = DagsterEvent(
+                event_type_value=DagsterEventType.PIPELINE_STARTING.value,
+                job_name=run.job_name,
+            )
+            instance.report_dagster_event(launch_started_event, run_id=run.run_id)
```

Comment:
moving it to QUEUED rather than STARTING would be more like the queued run coordinator that it will actually be using in most cases, would that change anything to make that change?

Reply:
nope, having it be QUEUED would be fine. It just needs to be a status other than NOT_STARTED so that the daemon doesn't hit the "the daemon died between creating the run and submitting the run path"